### ISSUE TITLE
`missing-const-for-fn` false positive with deref

### ISSUE THREAD

**Issue by icecream17** [rust-lang/rust-clippy#14747]
### Summary

When adding const according to the lint, it causes
```
error[E0015]: cannot perform non-const deref coercion on `<type>` in constant functions
```

This is different from #14020 because the deref is not unstably const, although #14020 seems to apply to stable...

### Lint Name

missing-const-for-fn

### Reproducer

I tried this code:

```rust
#![warn(clippy::missing_const_for_fn)]

fn repro_stable_only(vec: &Vec<usize>) -> &[usize] {
    vec
}
```

I saw this happen:

```
warning: this could be a `const fn`
 --> src/main.rs:4:1
  |
4 | / fn repro_stable_only(vec: &Vec<usize>) -> &[usize] {
5 | |     vec
6 | | }
  | |_^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#missing_const_for_fn
note: the lint level is defined here
 --> src/main.rs:2:9
  |
2 | #![warn(clippy::missing_const_for_fn)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: make the function `const`
  |
4 | const fn repro_stable_only(vec: &Vec<usize>) -> &[usize] {
  | +++++
```

I expected to see this happen:

No lint: [the deref is not const in stable or nightly](https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.deref). Weirdly it happens in stable but not nightly.

### Version

```text
rustc 1.86.0 (05f9846f8 2025-03-31)
binary: rustc
commit-hash: 05f9846f893b09a1be1fc8560e33fc3c815cfecb
commit-date: 2025-03-31
host: x86_64-unknown-linux-gnu
release: 1.86.0
LLVM version: 19.1.7
```

### Additional Labels

@rustbot label +I-suggestion-causes-error

**Comment by icecream17** [comment]
Oh, duplicate of https://github.com/rust-lang/rust-clippy/issues/14091
